Communitter/src/main/java/com/communitter/api/service/PostService.java

```diff
@@ -58,6 +49,18 @@ public String deletePost(Long id){
         return "Post deleted";
     }
 
+    @Transactional
+    public PostVote votePost(Long id, boolean isUpvote){
+        User author= (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```

Comment:
are you getting the authenticated user here?

Reply:
Yes I am